frontend/packages/client/src/components/UploadImageModal.js

```diff
@@ -0,0 +1,325 @@
+import React, { useCallback, useState, useEffect, useMemo } from "react";
+import { useDropzone } from "react-dropzone";
+import { useFileUploader } from "../hooks";
+import { Upload, Bin } from "../components/Svg";
+import { Loader } from "../components";
+
+const IMAGE_STATUS = {
+  notStarted: "not-started",
+  uploading: "uploading",
+  uploaded: "uploaded",
+};
+
+function ImageUploader({
+  image,
+  imageKey,
+  onUploaded,
+  deleteImage,
+  onUploadStared,
+}) {
+  const { uploadFile, loading, error } = useFileUploader({
+    useModalNotifications: false,
+  });
+  useEffect(() => {
+    async function upload(img) {
+      return uploadFile(img.file);
+    }
+    // image?.uploadStatus indicates upload did not started,
+    // then start upload and update status
+    if (image?.uploadStatus === IMAGE_STATUS.notStarted && !loading) {
+      onUploadStared(imageKey);
+      upload(image).then((uploaded) => {
+        if (error) {
+          // delete image if upload failed
+          deleteImage(imageKey);
+          return;
+        }
+        onUploaded(uploaded, imageKey);
+      });
+    }
+  }, [
+    image,
+    onUploaded,
+    loading,
+    error,
+    imageKey,
+    onUploadStared,
+    deleteImage,
+    uploadFile,
+  ]);
+
+  const fileName = image?.name ?? image.file.name;
```

Comment:
Not a necessary change, but if we are checking whether `image` exists here in the first term, I would think the check should be applied in the second term, as well.